scanner/src/test/kotlin/storages/ClearlyDefinedStorageTest.kt

```diff
@@ -0,0 +1,391 @@
+    val urlPath = "/harvest/$packageUrl"
+    val response = tools.joinToString(separator = ",", prefix = "[", postfix = "]") { "\"$it\"" }
+    wiremock.stubFor(
+        get(urlPathEqualTo(urlPath))
+            .withQueryParam("form", equalTo("list"))
+            .willReturn(
+                aResponse().withStatus(200)
+                    .withBody(response)
+            )
+    )
+}
+
+/**
+ * Stub a request for the harvested data from ScanCode for the given [packageUrl] on the [wiremock] server.
+ */
+private fun stubHarvestToolResponse(wiremock: WireMockServer, packageUrl: String) {
+    val urlPath = "/harvest/${toolUrl(packageUrl, "scancode", SCANCODE_VERSION)}"
+    wiremock.stubFor(
+        get(urlPathEqualTo(urlPath))
+            .withQueryParam("form", equalTo("streamed"))
+            .willReturn(
+                aResponse().withStatus(200)
+                    .withBodyFile(RESULT_FILE)
+            )
+    )
+}
+
+/**
+ * Check that the given [result] contains expected data.
+ */
+private fun assertValidResult(result: Result<ScanResultContainer>): ScanResult =
+    when (result) {
+        is Success -> {
+            result.result.id shouldBe TEST_IDENTIFIER
+            result.result.results shouldHaveSize 1
+            val scanResult = result.result.results[0]
+            scanResult.summary.licenseFindings.find {
+                it.location.path == TEST_PATH && it.license.licenses().contains("Apache-2.0")
+            }.shouldNotBeNull()
+            scanResult
+        }
+
+        is Failure -> fail("Expected success result, but got Failure(${result.error})")
+    }
+
+/**
+ * Check that the given [result] for package [expId] does not contain any data.
+ */
+private fun assertEmptyResult(result: Result<ScanResultContainer>, expId: Identifier = TEST_IDENTIFIER) {
+    when (result) {
+        is Success -> {
+            result.result.id shouldBe expId
+            result.result.results shouldHaveSize 0
+        }
+
+        is Failure -> fail("Unexpected result: $result")
+    }
+}
+
+/**
+ * Check whether the given [time] is close to the current time. This is used to check whether correct
+ * timestamps are set.
+ */
+private fun assertCurrentTime(time: Instant) {
+    val delta = Duration.between(time, Instant.now())
+    delta.isNegative shouldBe false
+    delta.compareTo(MAX_TIME_DELTA) shouldBeLessThan 0
+}
+
+class ClearlyDefinedStorageTest : WordSpec({
+    lateinit var wiremock: WireMockServer
+
+    beforeSpec {
+        wiremock = WireMockServer(
+            WireMockConfiguration.options()
+                .dynamicPort()
+                .usingFilesUnderDirectory("src/test/assets/")
+        )
+        wiremock.start()
+        WireMock.configureFor(wiremock.port())
+    }
+
+    afterSpec {
+        wiremock.stop()
+    }
+
+    beforeTest {
+        wiremock.resetAll()
+    }
+
+    "ClearlyDefinedStorage" should {
```

Comment:
Please add an empty line before the first assertion of each test.
...to get a clearer view on the 'arrange', 'act' 'assert' blocks.